monai/utils/misc.py

```diff
@@ -841,7 +841,7 @@ def _calculate_conversion_factor(self):
             return 1.0
         input_power = self._get_unit_power(self.input_unit)
         target_power = self._get_unit_power(self.target_unit)
-        self.conversion_factor = 10 ** (input_power - target_power)
+        return 10 ** (input_power - target_power)
```

Comment:
NOTE: In previous code the "self.conversion_factor" variable was never created or set if "if self.input_unit == self.target_unit:"